src/CoreText/CTFontManager.cs

```diff
@@ -122,8 +141,46 @@ public static NSError RegisterFontsForUrl (NSUrl fontUrl, CTFontManagerScope sco
 			}
 		}
 
+		[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+		public delegate bool CTFontRegistrationHandler (NSError[] errors, bool done);
+
+		internal delegate bool InnerRegistrationHandler (IntPtr block, IntPtr errors, bool done);
+		static readonly InnerRegistrationHandler callback = TrampolineRegistrationHandler;
+
+		[MonoPInvokeCallback (typeof (InnerRegistrationHandler))]
+		static unsafe bool TrampolineRegistrationHandler (IntPtr block, /* NSArray */ IntPtr errors, bool done)
+		{
+			var descriptor = (BlockLiteral *) block;
+			var del = (CTFontRegistrationHandler) (descriptor->Target);
+			return del != null ? del (NSArray.ArrayFromHandle<NSError> (errors), done) : true;
+		}
+
+		[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+		[DllImport (Constants.CoreTextLibrary)]
+		static extern unsafe void CTFontManagerRegisterFontURLs (/* CFArrayRef */ IntPtr fontUrls, CTFontManagerScope scope, bool enabled, void * registrationHandler);
+
+		[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+		public static void RegisterFonts (NSUrl [] fontUrls, CTFontManagerScope scope, bool enabled, CTFontRegistrationHandler registrationHandler)
```

Comment:
The block code can be simplified somewhat with this pattern: https://github.com/xamarin/xamarin-macios/blob/c99ad292b2e357f4f2acb083e03281a323da44a2/src/CoreFoundation/DispatchIO.cs#L56-L84

Here it would become something like this:

```csharp
[MonoPInvokeCallback (typeof (InnerRegistrationHandler))]
static bool TrampolineRegistrationHandler (IntPtr block, /* NSArray */ IntPtr errors, bool done)
{
	var del = BlockLiteral.GetTarget<CTFontRegistrationHandler> (block);
	return del != null ? del (NSArray.ArrayFromHandle<NSError> (errors), done) : true;
}

[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
[DllImport (Constants.CoreTextLibrary)]
static extern void CTFontManagerRegisterFontURLs (/* CFArrayRef */ IntPtr fontUrls, CTFontManagerScope scope, bool enabled, ref BlockLiteral registrationHandler);

[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
public static void RegisterFonts (NSUrl [] fontUrls, CTFontManagerScope scope, bool enabled, CTFontRegistrationHandler registrationHandler)
{
	using (var arr = EnsureNonNullArray (fontUrls, nameof (fontUrls))) {
		if (registrationHandler == null) {
			CTFontManagerRegisterFontURLs (arr.Handle, scope, enabled, null);
			return;
		}
		BlockLiteral block_handler = new BlockLiteral ();
		block_handler.SetupBlockUnsafe (callback, registrationHandler);
		CTFontManagerRegisterFontURLs (arr.Handle, scope, enabled, ref block_handler);
		block_ptr.CleanupBlock ();
	}
}
```

No unsafe code needed, and it's shorter, so easier to review and fewer lines to make mistakes in.